providers/default.rb

```diff
@@ -56,12 +56,44 @@ def create_haproxy_cfg
     owner 'root'
     group 'root'
     mode '0644'
-    notifies :reload, 'service[haproxy]', :delayed
+    notifies :reload, 'poise_service[haproxy]', :delayed
     variables(config: new_resource.config)
   end
 end
 
+def create_poise_service
+  poise_service_user node['haproxy']['user'] do
+    home '/home/haproxy'
+    group node['haproxy']['group']
+    action :create
+  end
+
+  node.override['haproxy']['conf_dir'] = ::File.join(node['haproxy']['install_method'].eql?('source') ? node['haproxy']['source']['prefix'] : '/', 'etc', 'haproxy')
```

Comment:
You have to override it if you call the provider from a wrapper, otherwise it seems to get the package one.